src/NuGet.Clients/NuGet.CommandLine/Program.cs

```diff
@@ -227,16 +226,55 @@ private void Initialize(CoreV2.NuGet.IFileSystem fileSystem, IConsole console)
         // This method acts as a binding redirect
         private Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
         {
-            var name = new AssemblyName(args.Name);
+            AssemblyName name = new AssemblyName(args.Name);
 
             if (string.Equals(name.Name, ThisExecutableName, StringComparison.OrdinalIgnoreCase))
             {
                 return typeof(Program).Assembly;
             }
+            // .NET Framework 4.x now triggers AssemblyResolve event for resource assemblies
+            // Catch this event for nuget.exe (NuGet.CommandLine) and NuGet.Command resource assemblies only
+            if (name.Name == "NuGet.resources" || name.Name == "NuGet.Commands.resources")
```

Comment:
Why do we need to special case NuGet.Commands.resources?

I imagine NuGet.resources is the NuGet.Commandline.resources one?

Reply:
I just followed a comment from that was part of dotnet list|add command implementation. Maybe @rrelyea has some more context here.

Reply:
Why aren't other assemblies like NuGet.Protocol, NuGet.Configuration, NuGet.Common, etc included as well?

Reply:
Because if we include all NuGet assemblies, this would result in a larger nuget.exe binary, 13 MB approximately.

Also, I think some core localization assemblies are shipe in nuget.localization package. Customers can use that package if they want.

Reply:
When a customer's machine uses a system language other than English, how can a customer using nuget.exe use NuGet.Localization to "fix" messages that are being output in English?

Reply:
Customers just need to unzip the package, and place nuget.exe alongside of lib\netstandard2.0 folder contents.

Reply:
> Customers just need to unzip the package, and place nuget.exe alongside of lib\netstandard2.0 folder contents.

Given that this affects UX, I feel like that needs discussed in a design spec.

Reply:
I'm not changing any new nuget.exe UX ; I'm just ILmerging as it was before. Why do we need a new UX spec? Do you want a UX design spec for merging all/some/none non-nuget.exe resource assemblies?

Reply:
My point is that if you want to tell customers to use `NuGet.Localization` then that's a UX change.

Reply:
Happy to set up an appointment.

Reply:
Just to clarify my comment. 

If you want to recommend that, then we need a spec. 

What prompted the whole discussion was the fact that it isn't obvious that we're only localizing nuget.commands and nuget.commandline only. 
The current implementation doesn't really localize strings from nuget.commands correctly, am that right?

NuGet.Protocol contains a healthy amount of strings that are aimed at the user and it's something that restore will frequently do. 
We should consider adding nuget.protocol as well imo.

Reply:
> The current implementation doesn't really localize strings from nuget.commands correctly, am that right?

That's correct.

> We should consider adding nuget.protocol as well imo.

We can do that in another PR :)

Reply:
> If you want to recommend that, then we need a spec.

I am going to enable support for nuget.exe localization; I just tried this approach of placing the satellite assemblies alongside nuget.exe and it didn't work. So, not recommended.